source/ui/beamline/AMCompositeControlDirectionMoveButton.cpp

```diff
@@ -0,0 +1,216 @@
+AMCompositeControlDirectionMoveButton::AMCompositeControlDirectionMoveButton(QWidget *parent,
+																			  AMControl* control, AMControl* control2, AMControl* controlR,
+																			  QList<double> stepSizes,
+																			  bool directionReversed, bool direction2Reversed ) :
+	AMControlMoveButton(parent, control, stepSizes, directionReversed)
+{
+	direction2Reversed_ = direction2Reversed;
+
+	control2_ = 0;
+	setControl2(control2);
+
+	controlR_ = 0;
+	setControlR(controlR);
+
+	checkStatus();
+
+	setToolTip("Composite Angle Move");
+
+}
+
+
+
+
+void AMCompositeControlDirectionMoveButton::setControl(AMControl *control)
+{
+	if(control == control_)
+		return;
+
+	if(directionMoveContextMenu_) {
+		directionMoveContextMenu_->deleteLater();
+		directionMoveContextMenu_ = 0;
+	}
+
+	if(control_) {
+		disconnect(control_, 0, this, 0);
+	}
+
+	control_ = control;
+	if(control_) {
+		connect(control_, SIGNAL(destroyed()), this, SLOT(oncontrolDestroyed()));
+		connect(control_, SIGNAL(connected(bool)), this, SLOT(checkStatus()));
+	}
+
+	setText(QString::number(currentStepSize()) % (control_ ? control_->units() : QString()));
+
+	// enabled / disabled:
+	checkStatus();
+}
+
+void AMCompositeControlDirectionMoveButton::setControl2(AMControl *control2)
+{
+	if(control2 == control2_)
+		return;
+
+	if(directionMoveContextMenu_) {
+		directionMoveContextMenu_->deleteLater();
+		directionMoveContextMenu_ = 0;
+	}
+
+	if(control2_) {
+		disconnect(control2_, 0, this, 0);
+	}
+
+	control2_ = control2;
+	if(control_) {
+		connect(control2_, SIGNAL(destroyed()), this, SLOT(onControl2Destroyed()));
+		connect(control2_, SIGNAL(connected(bool)), this, SLOT(checkStatus()));
+	}
+
+	setText(QString::number(currentStepSize()) % (control_ ? control_->units() : QString()));
+
+	// enabled / disabled:
+	checkStatus();
+}
+
+void AMCompositeControlDirectionMoveButton::setControlR(AMControl *controlR)
+{
+	if(controlR == controlR_)
+		return;
+
+	if(directionMoveContextMenu_) {
+		directionMoveContextMenu_->deleteLater();
+		directionMoveContextMenu_ = 0;
+	}
+
+	if(controlR_) {
+		disconnect(controlR_, 0, this, 0);
+	}
+
+	controlR_ = controlR;
+	if(control_) {
+		connect(controlR_, SIGNAL(destroyed()), this, SLOT(oncontrolRDestroyed()));
+		connect(controlR_, SIGNAL(connected(bool)), this, SLOT(checkStatus()));
+	}
+
+	setText(QString::number(currentStepSize()) % (control_ ? control_->units() : QString()));
+
+	// enabled / disabled:
+	checkStatus();
+}
+
+void AMCompositeControlDirectionMoveButton::onButtonClicked()
+{
+	if(control_ && control2_ && controlR_) {
+		control_->moveRelative(currentStepSize() * (directionReversed_ ? -1.0 : 1.0) * sin(controlR_->value()/57.297));
+		control2_->moveRelative(currentStepSize() * (direction2Reversed_ ? -1.0 : 1.0) * cos(controlR_->value()/57.297));
+	}
+
+//This seems that it should be better, but it not really any more "parallel", and it through "not within tolerance" errors when it is within tollerance.
```

Comment:
If you think that cruder, but working, two step move is okay then I'll axe the comment.  It seems like the list action is the "right" way to do it, but it has issues.